pyamg/util/utils.py

```diff
@@ -1192,6 +1192,21 @@ def unpack_arg(v):
         setup_smoother = getattr(relaxation.smoothing, 'setup_' + fn)
     except NameError:
         raise NameError("invalid presmoother method: ", fn)
+
+    # Estimate cost in WUs for different relaxation methods 
+    dcost = 1
+    if fn.endswith(('nr', 'ne')):
+        dcost *= 2
+    if 'sweep' in kwargs:
+        if kwargs['sweep'] == 'symmetric':
+            dcost *= 2
+    if 'iterations' in kwargs:
+        dcost *= kwargs['iterations']
+    if 'degree' in kwargs:
+        dcost *= kwargs['degree']
+
+    cost[0] += dcost
+
     # Get relaxation routine that takes only (A, x, b) as parameters
     relax = setup_smoother(lvl, **kwargs)
 
```

Comment:
See line 1362/1377 in this file (Git wouldn't let me comment that line). Think this is a bug I found today - filter_operator has an eliminate_zeros() at the end, but then the sparsity pattern of the returned matrix doesn't actually match that of the matrix 'graph' C passed in.